### PR TITLE
Add tests for `@time` compilation reporting

### PR DESCRIPTION
**Pull request by IanButterworth**
These tests are based off the examples in SnoopCompile's manual https://timholy.github.io/SnoopCompile.jl/stable/snoopr/

### CODE REVIEW

**Comment by IanButterworth** [comment]
How about using a file? (pushed)

**Comment by IanButterworth** [comment]
bump @vtjnash

**Review by vtjnash — COMMENTED** [review]
yeah, an IOStream should be fine, but let's use `mktemp` so it gets cleaned up immediately too

File: test/misc.jl [hunk 1]
```diff
@@ -354,6 +354,49 @@ after_comp, after_recomp = Base.cumulative_compile_time_ns() # no need to turn t
 
 end # redirect_stdout
 
+macro capture_stdout(ex)
+    quote
+        mktemp() do fname, f
+            redirect_stdout(f) do
+                $(esc(ex))
+            end
+            read(f, String)
```
Comment (IanButterworth) [inline test/misc.jl:363]:
```suggestion
            seekstart(f)
            read(f, String)
```

**Pull request merged**